File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int cnt=0,majority;
        for(int i=0;i<nums.size();i++) {
            if(cnt==0) {
                majority=nums[i];
                cnt=1;
            }else {
                if(majority==nums[i]) {
                    cnt++;
                }else {
                    cnt--;
                }
            }
        }
        return majority;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int ans=nums[0],count=1;
        for(int i=1;i<nums.length;i++) {
            if(count==0) {
                count++;
                ans=nums[i];
                continue;
            }
            if(ans==nums[i]) {
                count++;
            }else {
                count--;
            }
        }
        return ans;
    }
}